test/lit/passes/heap2local.wast

```diff
@@ -2640,24 +2640,34 @@
   (type $A (sub (struct (field (ref null $A)))))
   ;; CHECK:      (type $1 (func (result anyref)))
 
-  ;; CHECK:      (type $B (sub $A (struct (field (ref $A)))))
-  (type $B (sub $A (struct (field (ref $A)))))
+  ;; CHECK:      (type $B (sub $A (struct (field (ref $A)) (field i32))))
+  (type $B (sub $A (struct (field (ref $A)) (field i32))))
+
+  ;; CHECK:      (type $3 (func (result i32)))
 
   ;; CHECK:      (func $func (type $1) (result anyref)
   ;; CHECK-NEXT:  (local $a (ref $A))
   ;; CHECK-NEXT:  (local $1 (ref $A))
-  ;; CHECK-NEXT:  (local $2 (ref $A))
+  ;; CHECK-NEXT:  (local $2 i32)
+  ;; CHECK-NEXT:  (local $3 (ref $A))
+  ;; CHECK-NEXT:  (local $4 i32)
   ;; CHECK-NEXT:  (ref.cast (ref $B)
   ;; CHECK-NEXT:   (block (result (ref $A))
   ;; CHECK-NEXT:    (drop
   ;; CHECK-NEXT:     (block (result nullref)
-  ;; CHECK-NEXT:      (local.set $2
+  ;; CHECK-NEXT:      (local.set $3
   ;; CHECK-NEXT:       (struct.new $A
   ;; CHECK-NEXT:        (ref.null none)
   ;; CHECK-NEXT:       )
   ;; CHECK-NEXT:      )
+  ;; CHECK-NEXT:      (local.set $4
+  ;; CHECK-NEXT:       (i32.const 1)
+  ;; CHECK-NEXT:      )
   ;; CHECK-NEXT:      (local.set $1
-  ;; CHECK-NEXT:       (local.get $2)
+  ;; CHECK-NEXT:       (local.get $3)
+  ;; CHECK-NEXT:      )
+  ;; CHECK-NEXT:      (local.set $2
+  ;; CHECK-NEXT:       (local.get $4)
```

Comment:
These changes are what I meant in the second paragraph - `$B` has an extra field now, which is also handled here (with value `1`). This does not affect what the test is testing.